recipes/llm-voice-assistant/dotnet/LLMVoiceAssistant/Program.cs

```diff
@@ -0,0 +1,793 @@
+            listener.WakeWordDetected += speaker.OnStartWorkers;
+            listener.Start();
+            generator.Start();
+            Console.WriteLine($"Initialized. Say {config.PorcupineWakeWord} to start.");
+            while (true)
+            {
+                Thread.Sleep(1000);
+            }
+        }
+
+        static readonly string HELP_TEXT = @"
+Arguments:
+  --config: Path to a json config file to load the arguments from.
+
+  --access_key: `AccessKey` obtained from `Picovoice Console` (https://console.picovoice.ai/).
+
+  --picollm_model_path: Absolute path to the file containing LLM parameters (`.pllm`).
+
+  --keyword_model_path: Absolute path to the keyword model file (`.ppn`). If not set, `Picovoice` will be the wake phrase.
+
+  --cheetah_endpoint_duration_sec: Duration of silence (pause) after the user's utterance to consider it the end of the utterance.
+
+  --picollm_device: String representation of the device (e.g., CPU or GPU) to use for inference. Options:
+    - `best`: Picks the most suitable device.
+    - `gpu`: Uses the first available GPU device.
+    - `gpu:${GPU_INDEX}`: Selects a specific GPU by index.
+    - `cpu`: Runs on CPU with default threads.
+    - `cpu:${NUM_THREADS}`: Runs on CPU with specified threads.
+
+  --picollm_completion_token_limit: Maximum number of tokens in the completion. Set to `None` for no limit.
+
+  --picollm_presence_penalty: Positive value penalizes logits already appearing in the partial completion. `0.0` means no effect.
+
+  --picollm_frequency_penalty: Positive value penalizes logits based on frequency in partial completion. `0.0` means no effect.
+
+  --picollm_temperature: Sampling temperature controlling randomness:
+    - Higher value increases randomness.
+    - Lower value reduces variability.
+    - `0` selects the maximum logit.
+
+  --picollm_top_p: Limits sampling to high-probability logits forming the `top_p` portion of the probability mass (0 < `top_p` ≤ 1). `1.` turns off the feature.
+
+  --picollm_system_prompt: Text prompt to instruct the LLM before input.
+
+  --orca_warmup_sec: Duration of synthesized audio to buffer before streaming. Higher values help slower devices but increase initial delay.
+
+  --orca_speech_rate: Rate of speech for generated audio.
+
+  --porcupine_sensitivity: Sensitivity for detecting keywords.
+
+  --short_answers: Flag to enable short answers.
+
+  --profile: Flag to show runtime profiling information.
+";
+    }
+}
+
+interface Profiler
+{
+    public string Stats();
+}
+
+class RTFProfiler : Profiler
+{
+    private int _sampleRate;
+    private readonly Stopwatch _stopwatch = new Stopwatch();
+    private double _computeTimeSeconds = 0;
+    private double _audioTimeSeconds = 0;
+    public string Name { get; private set; }
+
+    public RTFProfiler(string name)
+    {
+        Name = name;
+    }
+
+    public void Init(int sampleRate)
+    {
+        _sampleRate = sampleRate;
+    }

```

Comment:
If you create the profiler in the actual worker classes then you can change the constructor to 
```csharp
public RTFProfiler(string name, int sampleRate)
```
and remove the `Init` function